### PR TITLE
COMMERCE-2231 Refactor commerce subscription entry

### PR DESCRIPTION
**Pull request by alessiorendina**

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test

File: commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceSubscriptionEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -73,6 +73,64 @@
 public class CommerceSubscriptionEntryLocalServiceImpl
 	extends CommerceSubscriptionEntryLocalServiceBaseImpl {
 
+	@Indexable(type = IndexableType.REINDEX)
+	@Override
+	public CommerceSubscriptionEntry addCommerceSubscriptionEntry(
+			long userId, long groupId, long commerceOrderItemId, int length,
```
Comment (ethanbustad) [inline commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceSubscriptionEntryLocalServiceImpl.java]: `int length` should really be `int subscriptionLength` to match the name of the field.

File: commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceSubscriptionEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -73,6 +73,64 @@
...
+
+		Date subscriptionStartDate =
+			cpSubscriptionType.getSubscriptionStartDate(
+				user.getTimeZone(), subscriptionTypeSettingsProperties);
+
+		commerceSubscriptionEntry.setStartDate(subscriptionStartDate);
+
+		commerceSubscriptionEntryPersistence.update(commerceSubscriptionEntry);
+
+		return commerceSubscriptionEntry;
```
Comment (ethanbustad) [inline commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceSubscriptionEntryLocalServiceImpl.java:132]: could just `return commerceSubscriptionEntryPersistence.update(commerceSubscriptionEntry);`

File: commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceSubscriptionEntryLocalServiceImpl.java (hunk 1 above)
Comment (marco-leo) [reply commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceSubscriptionEntryLocalServiceImpl.java]: @Commando18

File: commerce-service/src/main/java/com/liferay/commerce/internal/subscription/CommerceSubscriptionEntryHelperImpl.java [hunk 3]
```diff
@@ -57,12 +56,21 @@ public void checkCommerceSubscriptions(CommerceOrder commerceOrder)
...
-						commerceOrderItem.getCommerceOrderItemId());
+				CPSubscriptionInfo cpSubscriptionInfo =
+					cpInstance.getCPSubscriptionInfo();
+
+				if (cpSubscriptionInfo != null) {
+					_commerceSubscriptionEntryLocalService.
+						addCommerceSubscriptionEntry(
+							commerceAccount.getUserId(),
+							commerceOrder.getGroupId(),
+							commerceOrderItem.getCommerceOrderItemId(),
```
Comment (marco-leo) [inline commerce-service/src/main/java/com/liferay/commerce/internal/subscription/CommerceSubscriptionEntryHelperImpl.java:67]: @Commando18  Subscriptions can exist even with no orders

**Review by marco-leo — CHANGES_REQUESTED** [review]

**Comment by alessiorendina** [comment]
ci:test

**Comment by alessiorendina** [comment]
ci:test

**Pull request merged**